### PR TITLE
chore: migrate configs

### LINKED ISSUE #40: [CHORE] - Migrate Custom Configs
### What work needs to be performed?

The configurations for the tooling in this project need to be migrated to our custom global config packages.

The steps to do so will be:

- Fork and clone the repository, if you have not.
- Switch to a new branch:
  - `git checkout -b chore/migrate-configs`

## ESLint 
- Uninstall the root-level dependencies
  - `npm uninstall eslint-config-prettier eslint-plugin-import eslint-plugin-jsdoc eslint-plugin-no-only-tests eslint-plugin-prettier @typescript-eslint/eslint-plugin @typescript-eslint/parser`
- Install the eslint config
  - npm i --save-dev --save-exact @nhcarrigan/eslint-config
- Replace **all content** in the `.eslintrc.json` file with:
```json
{
  "extends": "@nhcarrigan"
}
```
- Add and commit changes
```bash
git add .
git commit -m "chore: migrate eslint"
```

## Prettier
- Install the prettier config
  - `npm i --save-dev --save-exact @nhcarrigan/prettier-config`
- Rename `.prettierrc` to `.prettierrc.json`
  - `git mv .prettierrc .prettierrc.json`
- Replace **all content** in the `.prettierrc.json` file with:
```json
"@nhcarrigan/prettier-config"
```
- Add and commit changes
```bash
git add .
git commit -m "chore: migrate prettier"
```

## TypeScript
- Install the typescript config
  - `npm i --save-dev --save-exact @nhcarrigan/typescript-config`
- Replace **all content** in the `tsconfig.json` file with:
```json
{
  "extends": "@nhcarrigan/typescript-config"
}
```
- Add and commit changes
```bash
git add .
git commit -m "chore: migrate typescript"
```

Push your changes and create a PR.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by at-the-vr**
# Pull Request

## Description:

- Migrate to custom configs
  - Remove bootstrapped dependencies
  - Add custom dependencies mentioned in Issue

## Related Issue:

Closes #40 

Note: This Repo did not included a `.prettierc` file (instead it already had a `.prettierc.json` included), so a couple of steps for Prettier were not applicable exactly

### CODE REVIEW

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -1,73 +1,3 @@
...
-        ],
-        "newlines-between": "always",
-        "alphabetize": {
-          "order": "asc",
-          "caseInsensitive": true
-        }
-      }
-    ]
-  }
+  "extends": "@nhcarrigan"
```
Comment (naomi-lgbt) [inline .eslintrc.json:2]:
```suggestion
  "extends": "@nhcarrigan",
  "rules": {
    "camelcase": [
      "error",
      {
        "allow": ["icon_url"]
      }
    ]
  }
```

Looks like we'll also need to allow an exception for the `icon_url` identifier.

File: .prettierrc.json [hunk 2]
```diff
@@ -1,5 +1,2 @@
-{
-  "endOfLine": "lf",
-  "useTabs": false,
-  "singleQuote": false
-}
+"@nhcarrigan/prettier-config"
+"@nhcarrigan/prettier-config"
```
[the remark this answers is not in the document]
Comment (at-the-vr) [reply .prettierrc.json]: Thanks for pointing that one (can't beleive that was there twice)… one thing have you hosted your configs on npm or somewhere (the way you import it from made me 🤔)

[the remark this answers is not in the document]
Comment (naomi-lgbt) [reply .prettierrc.json]: Yep, they're all hosted via [npm](https://www.npmjs.com/package/@nhcarrigan/eslint-config).

**Review by naomi-lgbt — APPROVED** [review]
Nicely done~!

**Comment by naomi-lgbt** [comment]
## Congratulations

Congratulations on making your first pull request to our project. We have reviewed your contributions and are happy to accept them.

We hope you enjoyed the contribution process, and encourage you to watch for issues labelled `help wanted` for additional opportunities to contribute.

You can also see all open issues [through our contributor tool](https://contribute.naomi.lgbt)

**Pull request merged**